inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -1184,4 +1197,57 @@ public File readKbFileFromClassPathResource(String aLocation) throws IOException
         return kbFile;
     }
 
+
+    private IndexWriter getIndexWriter(KnowledgeBase aKb) throws IOException
+    {
+        Analyzer analyzer = new StandardAnalyzer();
+        Directory directory = FSDirectory
+            .open(new File(luceneIndexDir, aKb.getRepositoryId()).toPath());
+        return new IndexWriter(directory, new IndexWriterConfig(analyzer));
+    }
+
+    @Override
+    public void indexLocalKb(KnowledgeBase aKb)
```

Comment:
There is already `LuceneSail.reindex()`. I expect it should be possible to do something like `((LuceneSail) ((SailRepository) repoManager.getRepository(kb.getRepositoryId())).getSail()).reindex()` - obviously not like that but rather with a could of "if / instanceof" checks along the way.